charts/thanos/templates/query-frontend/serviceaccount.yaml

```
@@ -1,4 +1,4 @@
-{{- if and .Values.queryFrontend.enabled .Values.queryFrontend.serviceAccount.create -}}
+{{- if and .Values.query.enabled .Values.queryFrontend.enabled .Values.queryFrontend.serviceAccount.create -}}
```

```suggestion
{{- if and .Values.query.enabled (and .Values.queryFrontend.enabled .Values.queryFrontend.serviceAccount.create) -}}
```
Query disabled should logically short circuit other checks.